### PR TITLE
Add German to transition words assessment

### LINKED ISSUE #734: Transition words for German
We should put together a transition word list for German to use in the transition word assessment.

### PR DESCRIPTION
**Pull request by IreneStr**
- Adds German transition words and two-part transition words.
- Adds getLanguage helper to get language part of the locale.
- Adds transitionWordsLocale helper that provides the correct transition word lists based on the paper's locale.
- Moves language-specific word lists to language directories in researches directory.
- Adds locale input field to browserified example.

For testing: Change locale to de_DE. Write a sentence with a German transition word and a sentence with a German two-part transition word.  

Fixes #734

### CODE REVIEW

File: js/assessments/transitionWordsAssessment.js [hunk 1]
```diff
@@ -106,7 +106,9 @@ module.exports = {
 	identifier: "textTransitionWords",
 	getResult: transitionWordsAssessment,
 	isApplicable: function( paper ) {
-		return ( paper.getLocale().indexOf( "en_" ) > -1 && paper.hasText() );
+		var locale = paper.getLocale().split( "_" )[ 0 ];
```
Comment (atimmer) [inline js/assessments/transitionWordsAssessment.js]: Please use `getLanguage` here.

File: js/assessments/transitionWordsAssessment.js [hunk 2]
```diff
@@ -106,7 +106,9 @@ module.exports = {
...
+		var validLocale = locale === "en" || locale === "de";
+		return ( validLocale > -1 && paper.hasText() );
```
Comment (atimmer) [inline js/assessments/transitionWordsAssessment.js]: `validLocale` is already a boolean so there is no need for the `> -1` part

File: js/helpers/transitionWordsLocale.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+var transitionWordsEnglish = require( "../researches/english/transitionWords.js" );
+var twoPartTransitionWordsEnglish = require( "../researches/english/twoPartTransitionWords.js" );
+var transitionWordsGerman = require( "../researches/german/transitionWords.js" );
+var twoPartTransitionWordsGerman = require( "../researches/german/twoPartTransitionWords.js" );
+
+var getLanguage = require( "./getLanguage.js" );
+
+module.exports = function( locale ) {
```
Comment (atimmer) [inline js/helpers/transitionWordsLocale.js]: Please call this file `getTransitionWords`.

File: js/researches/findTransitionWords.js [hunk 4]
```diff
@@ -1,54 +1,56 @@
-var transitionWords = require( "../config/transitionWords.js" );
-var twoPartTransitionWords = require( "../config/twoPartTransitionWords.js" );
 var createRegexFromDoubleArray = require( "../stringProcessing/createRegexFromDoubleArray.js" );
 var getSentences = require( "../stringProcessing/getSentences.js" );
 var matchWordInSentence = require( "../stringProcessing/matchWordInSentence.js" );
 var normalizeSingleQuotes = require( "../stringProcessing/quotes.js" ).normalizeSingle;
+var transitionWordsLocale = require( "../helpers/transitionWordsLocale.js" );
```
Comment (atimmer) [inline js/researches/findTransitionWords.js]: Please rename this to `getTransitionWords`

File: js/values/Paper.js [hunk 5]
```diff
@@ -35,6 +35,9 @@ var Paper = function( text, attributes ) {
 
 	attributes = attributes || {};
 	defaults( attributes, defaultAttributes );
+	if ( attributes.locale === "" ) {
+		attributes.locale = defaultAttributes.locale;
```
Comment (atimmer) [inline js/values/Paper.js]: This line is not covered by the specs.

File: spec/helpers/getLanguageSpec.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+var getLanguage = require( "../../js/helpers/getLanguage.js" );
+
+describe("a function to get the language from the locale ", function(){
+	it("returns en in case of en_US", function(){
+		expect( getLanguage( 'en_US' ) ).toBe( 'en' );
+	});
+	it("returns de in case of de_DE", function(){
+		expect( getLanguage( 'de_DE' ) ).toBe( 'de' );
+	});
+});
```
Comment (atimmer) [inline spec/helpers/getLanguageSpec.js]: Please add a spec for the expected result when passing an empty string.

**Comment by atimmer** [comment]
CR Done 👷🏻

**Comment by atimmer** [comment]
CR Done 👍🏻

**Comment by boblinthorst** [comment]
Acceptance ok 👍

**Pull request merged**